src/Discord.Net.WebSocket/DiscordSocketClient.cs

```diff
@@ -242,14 +261,27 @@ private async Task OnConnectingAsync()
                 else
                 {
                     await _gatewayLogger.DebugAsync("Identifying").ConfigureAwait(false);
-                    await ApiClient.SendIdentifyAsync(shardID: ShardId, totalShards: TotalShards, guildSubscriptions: _guildSubscriptions).ConfigureAwait(false);
+                    if (Activity is RichGame)
+                        throw new NotSupportedException("Outgoing Rich Presences are not supported via WebSocket.");
+                    await ApiClient.SendIdentifyAsync(shardID: ShardId, totalShards: TotalShards, guildSubscriptions: _guildSubscriptions, presence: new StatusUpdateParams
+                    {
+                        Game = Activity == null ? null : new GameModel
+                        {
+                            Name = Activity.Name,
+                            Type = Activity.Type,
+                            StreamUrl = Activity is StreamingGame sg ? sg.Url : null,
+                        },
+                        Status = Status,
+                        IsAFK = Status == UserStatus.AFK,
+                        IdleSince = _statusSince != null ? _statusSince.Value.ToUnixTimeMilliseconds() : (long?)null
+                    }).ConfigureAwait(false);
                 }
 
                 //Wait for READY
                 await _connection.WaitAsync().ConfigureAwait(false);
 
-                await _gatewayLogger.DebugAsync("Sending Status").ConfigureAwait(false);
-                await SendStatusAsync().ConfigureAwait(false);
+                /*await _gatewayLogger.DebugAsync("Sending Status").ConfigureAwait(false);
+                await SendStatusAsync().ConfigureAwait(false);*/
```

Comment:
Why is this commented out?

Reply:
```suggestion
if (Activity is RichGame)
                        throw new NotSupportedException("Outgoing Rich Presences are not supported via WebSocket.");
                    await ApiClient.SendIdentifyAsync(shardID: ShardId, totalShards: TotalShards, guildSubscriptions: _guildSubscriptions, presence: new StatusUpdateParams
                    {
                        Game = Activity == null ? null : new GameModel
                        {
                            Name = Activity.Name,
                            Type = Activity.Type,
                            StreamUrl = Activity is StreamingGame sg ? sg.Url : null,
                        },
                        Status = Status,
                        IsAFK = Status == UserStatus.AFK,
                        IdleSince = _statusSince != null ? _statusSince.Value.ToUnixTimeMilliseconds() : (long?)null
                    }).ConfigureAwait(false);
```
Same thing but on the way when sending the identify event, Discord supports setting status on identify event

Reply:
@davidsl4 uh might want to rebase this pr to fix the merge conflicts.